### PR TITLE
Removed instant selectors (Fixes Issue #1377)

### LINKED ISSUE #1377: Remove instant selectors
Another issue from the dev meetings, @orbeckst proposed removing these selectors

I'll try and summarise the currently available operations and what they do, along with the best alternative

```python
u = mda.Universe()

# AtomGroup
# attribute based
u.atoms.H
u.atoms.select_atoms('name H')
# getitem based
u.atoms['H']
u.atoms.select_atoms('name H')

# ResidueGroup
rg = u.residues
rg.LYS
rg[rg.resname == 'LYS']

# Segment
u.segments[0].r1  # gives first residue
u.segments[0].residues[0]

# Universe
u.SYSTEM  # based on segids in topology
u.segments[u.segments.segids == 'SYSTEM']
```

They're also a little weird in that they **usually* return Groups, but sometimes return singular objects if only one object was found.  Ie `RG.LYS` will return a `ResidueGroup` unless there's only one LYS, in which case you get a `Residue`

Should we ditch (all of?) these?

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1377

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/groups.py [hunk 1]
```diff
@@ -2203,55 +2203,19 @@ class AtomGroup(GroupBase):
...
     --------
     :class:`MDAnalysis.core.universe.Universe`
 
-
     .. deprecated:: 0.16.2
        *Instant selectors* of :class:`AtomGroup` will be removed in the 1.0
-       release. See :ref:`Instant selectors <instance-selectors>` for details
-       and alternatives.
+       release.
+    .. versionchanged:: 1.0
```
Comment (RMeli) [inline package/MDAnalysis/core/groups.py]: A bit of a pedantic comment, but should `versionchanged` also include the patch version?

**Comment by richardjgowers** [comment]
@IAlibay thanks!

**Comment by richardjgowers** [comment]
@MDAnalysis/coredevs needs a review

File: package/MDAnalysis/core/groups.py (hunk 1 above)
Comment (orbeckst) [reply package/MDAnalysis/core/groups.py]: Generally yes

Comment (orbeckst) [reply package/MDAnalysis/core/groups.py]: yes

File: package/MDAnalysis/core/groups.py [hunk 2]
```diff
@@ -3256,8 +3214,8 @@ class ResidueGroup(GroupBase):
 
     .. deprecated:: 0.16.2
        *Instant selectors* of Segments will be removed in the 1.0 release.
-       See :ref:`Instant selectors <instance-selectors>` for details and
-       alternatives.
+    .. versionchanged:: 1.0
```
Comment (orbeckst) [inline package/MDAnalysis/core/groups.py]: 1.0.0 ;-)

File: package/MDAnalysis/core/universe.py [hunk 3]
```diff
@@ -369,8 +315,9 @@ class Universe(object):
     bonds, angles, dihedrals
         master ConnectivityGroups for each connectivity type
 
-    .. versionchanged:: 0.21.0
+    .. versionchanged:: 1.0
```
Comment (orbeckst) [inline package/MDAnalysis/core/universe.py]: 1.0.0

**Pull request merged**